server/runtime.go

```diff
@@ -37,16 +40,111 @@ func (s *Server) Version(ctx context.Context, req *pb.VersionRequest) (*pb.Versi
 func (s *Server) CreatePodSandbox(ctx context.Context, req *pb.CreatePodSandboxRequest) (*pb.CreatePodSandboxResponse, error) {
 	var err error
 
-	// TODO: Parametrize as a global argument to ocid
-	ocidSandboxDir := "/var/lib/ocid/sandbox"
-	podSandboxDir := filepath.Join(ocidSandboxDir, req.GetConfig().GetName())
+	// process req.Name
+	name := req.GetConfig().GetName()
+	if name == "" {
+		d, err := ioutil.TempDir(s.sandboxDir, "")
+		if err != nil {
+			return nil, err
+		}
+		name = filepath.Base(d)
+	}
+	podSandboxDir := filepath.Join(s.sandboxDir, name)
+	if err := os.MkdirAll(podSandboxDir, 0755); err != nil {
+		return nil, err
+	}
 
+	// creates a spec Generator with the default spec.
 	g := generate.New()
 
-	// TODO: Customize the config per the settings in the req
+	// process req.Hostname
+	hostname := req.GetConfig().GetHostname()
+	if hostname != "" {
+		g.SetHostname(hostname)
+	}
+
+	// process req.LogDirectory
+	logDir := req.GetConfig().GetLogDirectory()
+	if logDir == "" {
+		logDir = "/var/log/ocid/pods"
+	}
+	fmt.Println(logDir)
+
+	// TODO(hmeng): how to utilize this info?
```

Comment:
We would most likely need to create resolv.conf on the host and bind mount it to /etc/resolv.conf inside the container. So, this will mean adding a bind mount. (I need to confirm this.)